### PR TITLE
Support Oracle database connection

### LINKED ISSUE #54: Support Oracle database connection
The issue is similar to #48 but requesting add support for Oracle database connection. Here is the mockup design for the feature:

image

### PR DESCRIPTION
**Pull request by majguo**
## Description

The PR is to resolve the following issues:
* Resolve #54 

## Change summary

* Add a new database type **Oracle database** in blade **Database**.
* Update template to handle Oracle database connection.
* Add new scripts to create Oracle JDBC provider and data source.
* Update the main `install.sh` to handle the Oracle database connection.
* Update the `integration-test` pipeline to support Oracle database connection.
* Update setup and tear-down scripts to handle the new GitHub action secret `ORACLE_DB_PASSWORD`.

## Testing

Verified the following 3 test cases with `integration-test` pipeline:
* Oracle database enabled: [integration-test-71](https://github.com/majguo/azure.websphere-traditional.singleserver/actions/runs/3286504940)
* Db2 database enabled: [integration-test-72](https://github.com/majguo/azure.websphere-traditional.singleserver/actions/runs/3286566633)
* Database connection disabled: [integration-test-73](https://github.com/majguo/azure.websphere-traditional.singleserver/actions/runs/3286567553)

If you want to enable Oracle db connection in `integration-test` pipeline, pls look at #56.

These three cases are also verified using the [preview link](https://portal.azure.com/?feature.customportal=false#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewwas-single-server). Feel free to do a live testing with the [preview link](https://portal.azure.com/?feature.customportal=false#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewwas-single-server).

## UI changes review

Besides visiting [preview link](https://portal.azure.com/?feature.customportal=false#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewwas-single-server) for reviewing the new database type `Oracle database` in blade `Database`, reviewer can preview the UI changes using the **Create UI Definition Sandbox**:

- Open [Create UI Definition Sandbox](https://portal.azure.com/?feature.customPortal=false#blade/Microsoft_Azure_CreateUIDef/SandboxBlade) in a new tab of browser.
- Open UI source code [createUiDefinition.json](https://raw.githubusercontent.com/majguo/azure.websphere-traditional.singleserver/oracle-db-conn/src/main/arm/createUiDefinition.json) in a new tab of the browser.
- Replace the content of work area in **Create UI Definition Sandbox** with the content of **UI source code `createUiDefinition.json`**.
- Click **Preview** at the left-bottom of **Create UI Definition Sandbox**.
- The UI changes in the existing `Database` tab include:
   - Add a new database type `Oracle database`.
   - The data source connection string example for `Oracle database` is updated accordingly. 

Here is the screenshot of UI changes overview:
image

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -102,31 +104,60 @@ jobs:
...
+            --vnet-name ${{ env.dbVmName }}VNET --enable-auto-update false \
+            --tags SkipASMAzSecPack=true SkipNRMSCorp=true SkipNRMSDatabricks=true SkipNRMSDB=true SkipNRMSHigh=true SkipNRMSMedium=true SkipNRMSRDPSSH=true SkipNRMSSAW=true SkipNRMSMgmt=true
+          az vm disk attach --name oradata01 --new --resource-group ${{ env.testResourceGroup }} --vm-name ${{ env.dbVmName }} --size-gb 64 --sku StandardSSD_LRS
+          az vm open-port -g ${{ env.testResourceGroup }} -n ${{ env.dbVmName }} --port 1521,5502 --priority 100
+          az vm extension set --name CustomScript \
+            --extension-instance-name install-oracle \
+            --resource-group ${{ env.testResourceGroup }} --vm-name ${{ env.dbVmName }} \
+            --publisher Microsoft.Azure.Extensions --version 2.0 \
+            --settings "{\"fileUris\": [\"${{ env.scriptLocation }}install-oracle-main.sh\", \"${{ env.scriptLocation }}install-oracle.sh\"]}" \
+            --protected-settings "{\"commandToExecute\":\"bash install-oracle-main.sh ${{ env.oracleDBPassword }}\"}"
```
Comment (edburns) [inline .github/workflows/integration-test.yaml:138]: Amazing stuff here.

File: src/main/scripts/create-ds-db2.py.template [hunk 2]
```diff
@@ -39,8 +39,8 @@ j2cUser = AdminConfig.create('JAASAuthData', security, jaasAttrs)
 newjdbc = AdminConfig.getid('/JDBCProvider:DB2JDBCProvider/')
 name = ['name', '${DB2_DATASOURCE_NAME}']
 jndi = ['jndiName', '${DB2_DATASOURCE_JNDI_NAME}']
-auth = ['authDataAlias' , userAlias]
-authMechanism = ['authMechanismPreference' , 'BASIC_PASSWORD']
+auth = ['authDataAlias', userAlias]
+authMechanism = ['authMechanismPreference', 'BASIC_PASSWORD']
```
Comment (edburns) [inline src/main/scripts/create-ds-db2.py.template:43]: Good catch.

File: src/main/scripts/create-ds.sh [hunk 3]
```diff
@@ -57,6 +59,19 @@ if [ $dbType == "db2" ]; then
     sed -i "s#\${DB2_DATASOURCE_JNDI_NAME}#${jdbcDSJNDIName}#g" $createDsScript
     sed -i "s/\${DB2_SERVER_NAME}/${db2ServerName}/g" $createDsScript
     sed -i "s/\${PORT_NUMBER}/${db2ServerPortNumber}/g" $createDsScript
+elif [ $dbType == "oracle" ]; then
+    # Download jdbc drivers
+    curl -Lo ${jdbcDriverPath}/ojdbc8.jar https://download.oracle.com/otn-pub/otn_software/jdbc/1916/ojdbc8.jar
+    jdbcDriverClassPath=$(realpath "$jdbcDriverPath"/ojdbc8.jar)
```
Comment (edburns) [inline src/main/scripts/create-ds.sh:65]: I'm sorry to do this, but I need you to put this inside a retry loop.

File: src/main/scripts/install.sh [hunk 4]
```diff
@@ -112,7 +112,7 @@ adminPassword=$2
 enableDB=$3
 dbType=$4
 jdbcDSJNDIName=$5
-dsConnectionURL=$6
+dsConnectionString=$6
```
Comment (edburns) [inline src/main/scripts/install.sh:115]: I am aware you know this, but please remember this query. https://devdiv.visualstudio.com/DevDiv/_queries/query/2625d14f-0deb-4763-b4f5-e0495204c329/

**Review by edburns — CHANGES_REQUESTED** [review]
Need retry loop in src/main/scripts/create-ds.sh.

File: .github/workflows/integration-test.yaml (hunk 1 above)
Comment (majguo) [reply .github/workflows/integration-test.yaml:138]: I haven't looked into it as Galia shared me an [official doc](https://learn.microsoft.com/en-us/azure/virtual-machines/workloads/oracle/oracle-database-quick-create) from `learn.microsoft.com` which installs an Oracle server on an Azure VM. 
Thanks for sharing [oracle container registry database image](https://container-registry.oracle.com/ords/f?p=113:4:16393593035915:::4:P4_REPOSITORY,AI_REPOSITORY,AI_REPOSITORY_NAME,P4_REPOSITORY_NAME,P4_EULA_ID,P4_BUSINESS_AREA_ID:8,8,Oracle%20Database%20Standard%20Edition%202,Oracle%20Database%20Standard%20Edition%202,1,0&cs=3p0HpayI5XMUmDL4HWFy3lBevy7cDiY4UNmKz6wbqcfnCA4yncP5KlQZq1KLHHdgBfzJREF2VhTwXVGxtsRroQQ), it looks like another option to install Oracle server.

File: src/main/scripts/install.sh (hunk 4 above)
Comment (majguo) [reply src/main/scripts/install.sh:115]: Good catch. However, env vars are supported by `Microsoft.Resources/deploymentScripts` but not supported for `Microsoft.Compute/virtualMachines/extensions` as for now.

File: src/main/scripts/create-ds.sh (hunk 3 above)
Comment (majguo) [reply src/main/scripts/create-ds.sh]: Good point. Fixed in [85a7df9](https://github.com/WASdev/azure.websphere-traditional.singleserver/pull/55/commits/85a7df970a16d683b442045a943a757260818945).

**Comment by majguo** [comment]
> Need retry loop in src/main/scripts/create-ds.sh.

Thanks for review. It's addressed in [85a7df9](https://github.com/WASdev/azure.websphere-traditional.singleserver/pull/55/commits/85a7df970a16d683b442045a943a757260818945).

**Review by edburns — APPROVED** [review]

**Pull request merged**